providers/google/src/airflow/providers/google/cloud/hooks/bigquery.py

```diff
@@ -1285,10 +1292,17 @@ def generate_job_id(self, job_id, dag_id, task_id, logical_date, configuration,
         if job_id:
             return f"{job_id}_{uniqueness_suffix}"
 
-        exec_date = logical_date.isoformat()
+        exec_date = date.isoformat()
         job_id = f"airflow_{dag_id}_{task_id}_{exec_date}_{uniqueness_suffix}"
         return re.sub(r"[:\-+.]", "_", job_id)
 
+    def get_exec_date(self, context: Context) -> pendulum.DateTime:
```

Comment:
This method name is not accurate. `get_logical_date_or_run_after` might be better